### ISSUE TITLE
On Factorio 2.0 server crashes randomly after using the planner

### ISSUE THREAD

**Issue by SamueleLorefice** [ClaudeMetz/FactoryPlanner#353]
### Problem Description

Can't sort out exactly what triggers the crash, but after using it, few minutes later the server will crash with 

```The mod Factory Planner (2.0.1) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::on_nth_tick(893801)
__factoryplanner__/util/context.lua:109: attempt to index local 'object' (a nil value)
stack traceback:
	__factoryplanner__/util/context.lua:109: in function 'remove'
	__factoryplanner__/ui/main/factory_list.lua:7: in function 'handler'
	__factoryplanner__/util/nth_tick.lua:10: in function <__factoryplanner__/util/nth_tick.lua:7>```
	
```

**Comment by ClaudeMetz** [comment]
Hm that's a tricky one without a reproduction. Maybe it's multiplayer latency related :/ I'll try a fix.

**Comment by LordZepp** [comment]
> Hm that's a tricky one without a reproduction. Maybe it's multiplayer latency related :/ I'll try a fix.

I have the same problem in solo world. Crashes in 10 seconds every time now.

The mod Factory Planner (2.0.1) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::on_nth_tick(1813512)
__factoryplanner__/util/context.lua:109: attempt to index local 'object' (a nil value)
stack traceback:
	__factoryplanner__/util/context.lua:109: in function 'remove'
	__factoryplanner__/ui/main/factory_list.lua:7: in function 'handler'
	__factoryplanner__/util/nth_tick.lua:10: in function <__factoryplanner__/util/nth_tick.lua:7>

Mods:
![image](https://github.com/user-attachments/assets/c0ccd207-951a-478e-85b8-37532f3eddef)

Save:
[Space Age.zip](https://github.com/user-attachments/files/17471588/Space.Age.zip)

**Comment by ClaudeMetz** [comment]
Sorry about this one, it happens when you deleted a District where you recently deleted some factories. Thanks for the reports, fixed for 2.0.2 which is due to be released soon!
